### PR TITLE
Add elapsed return value to select modules

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
It can be quite useful to know exactly how much time has elapsed downloading/waiting. This improves existing modules or updates documentation.

- This PR brings feature parity to **get_url** (from win_get_url).
- We ensure `elapsed` is always returned (also when it eventually fails/times out)
- We added `elapsed` to **uri** and **win_uri** (cfr. get_url) on the basis that it helps to troubleshoot network-related issues (which can happen at any time, so `--debug` or altering strategy is no option here).

This relates to #37957

##### ISSUE TYPE
 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME
get_url, uri, wait_for, win_get_url, win_reboot, win_uri

##### ANSIBLE VERSION
v2.6

### CODE REVIEW

File: lib/ansible/modules/windows/win_uri.py [hunk 1]
```diff
@@ -178,6 +178,11 @@
 '''
 
 RETURN = r'''
+elapsed:
+  description: The number of seconds that elapsed while performing the download
+  returned: always
+  type: int
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_uri.py]: This is actually a decimal (not sure what we call it in Ansible, float?) and can return up to 7 decimal places IIRC.

Comment (dagwieers) [reply lib/ansible/modules/windows/win_uri.py]: Then we have it mispresented for win_reboot too :-)
Will fix !

**Comment by dagwieers** [comment]
@mattclay Can we try this ? It may help us find the cause for #37957

**Comment by dagwieers** [comment]
I now have built this 4 times without any issues, so I cannot reproduce the problem. But as soon as we can, we should have a clearer indication whether it is related to different timeout values.

Should we also look at more detailed error information ?

**Comment by dagwieers** [comment]
So the last run before going to bed triggers it for a single test.
The elapsed time for the HTTP request to timeout was **5.7812563**.

But what sticks out is the fact there is a **HTTP 408** status code which means the server closed the connection before the request was served.

Why is httpbin closing the connection so quickly ?

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/64599/29/tests

**Comment by dagwieers** [comment]
@jborean93 @nitzmahone I'd like this one to go forward as well.

**Comment by jhawkesworth** [comment]
How did I miss this before.
Code looks good to me.

**Review by jhawkesworth — APPROVED** [review]

**Comment by dagwieers** [comment]
@mattclay @jborean93 Is this a known issue ? I get UNSTABLE and FAILED for this task, never seem to get past this with message: **timeout while waiting for string regex file c.\* in file C:\\ansible\\win_wait_for\\test.txt to match**.
```
2018-07-30 08:49:24 TASK [win_wait_for : wait for file contents to match regex - empty file] *******
2018-07-30 08:49:24 task path: /root/ansible/test/integration/targets/win_wait_for/tasks/main.yml:157
2018-07-30 08:49:24 Using module file /root/ansible/lib/ansible/modules/windows/win_wait_for.ps1
2018-07-30 08:49:24 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> ESTABLISH WINRM CONNECTION FOR USER: administrator on PORT 5985 TO ec2-18-220-171-10.us-east-2.compute.amazonaws.com
2018-07-30 08:49:24 checking if winrm_host ec2-18-220-171-10.us-east-2.compute.amazonaws.com is an IPv6 address
2018-07-30 08:49:24 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM CONNECT: transport=ntlm endpoint=http://ec2-18-220-171-10.us-east-2.compute.amazonaws.com:5985/wsman
2018-07-30 08:49:24 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM OPEN SHELL: 623B8FAF-EA3C-4092-8570-6F8A6324BB98
2018-07-30 08:49:24 EXEC (via pipeline wrapper)
2018-07-30 08:49:24 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM EXEC 'PowerShell' ['-NoProfile', '-NonInteractive', '-ExecutionPolicy', 'Unrestricted', '-']
2018-07-30 08:54:26 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM RESULT u'<Response code 0, out "{"changed":false,"el", err "">'
2018-07-30 08:54:26 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM STDOUT {"changed":false,"elapsed":300.6833185,"msg":"timeout while waiting for string regex file c.* in file C:\\ansible\\win_wait_for\\test.txt to match","failed":true,"wait_attempts":296}

2018-07-30 08:54:26 
2018-07-30 08:54:26 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM STDERR 
2018-07-30 08:54:26 <ec2-18-220-171-10.us-east-2.compute.amazonaws.com> WINRM CLOSE SHELL: 623B8FAF-EA3C-4092-8570-6F8A6324BB98
2018-07-30 08:54:26 fatal: [windows-2016]: FAILED! => {
2018-07-30 08:54:26     "changed": false, 
2018-07-30 08:54:26     "elapsed": 300.6833185, 
2018-07-30 08:54:26     "msg": "timeout while waiting for string regex file c.* in file C:\\ansible\\win_wait_for\\test.txt to match", 
2018-07-30 08:54:26     "wait_attempts": 296
2018-07-30 08:54:26 }
```

**Comment by dagwieers** [comment]
Another attempt worked, after going to the original codebase (not the current).

**Comment by jborean93** [comment]
@dagwieers the win_wait_for tests are unstable due to https://github.com/ansible/ansible/issues/31030. For some reason on the cloud hosts the async task starts but the subsequent win_shell/command operation fails to start the executable and Ansible does not expect this.

**Comment by dagwieers** [comment]
Calling out to @AnderEnder @gregswift @jarv @jhoekx @jpmens @romeotheriault @blakfeld
Any objections to these changes to your module ?

*(This change was already approved during a Windows WG meeting)*

**Comment by jpmens** [comment]
shipit

**Review by jborean93 — APPROVED** [review]

**Comment by dagwieers** [comment]
So now that v2.7 is branched, we can move this forward.

**Pull request merged**